### ISSUE TITLE
Exception with specific versions of Parser - appears to be Parser bug

### ISSUE THREAD

**Issue by cschneid** [scoutapp/scout_apm_ruby#303]
A customer ran into an exception when using AutoInstruments, with Parser 2.6.3.0, where it errors out during loading with: 

```
  bundle exec rake assets:precompile
/usr/local/bundle/gems/builder-3.1.4/lib/builder/xchar.rb:111: warning: constant ::Fixnum is deprecated
warning: parser/current is loading parser/ruby25, which recognizes
warning: 2.5.5-compliant syntax, but you are running 2.5.3.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
rake aborted!
NameError: uninitialized constant Parser::Base
/usr/local/bundle/gems/parser-2.6.3.0/lib/parser/ruby25.rb:14:in `<module:Parser>'
/usr/local/bundle/gems/parser-2.6.3.0/lib/parser/ruby25.rb:13:in `<top (required)>'
/usr/local/bundle/gems/parser-2.6.3.0/lib/parser/current.rb:65:in `<module:Parser>'
/usr/local/bundle/gems/parser-2.6.3.0/lib/parser/current.rb:3:in `<top (required)>'
/usr/local/bundle/gems/scout_apm-2.6.2/lib/scout_apm/auto_instrument/parser.rb:2:in `<top (required)>'
/usr/local/bundle/gems/scout_apm-2.6.2/lib/scout_apm/auto_instrument/rails.rb:3:in `<top (required)>'
/usr/local/bundle/gems/scout_apm-2.6.2/lib/scout_apm/auto_instrument/instruction_sequence.rb:2:in `<top (required)>'
/usr/local/bundle/gems/scout_apm-2.6.2/lib/scout_apm/auto_instrument.rb:2:in `<top (required)>'
/usr/local/bundle/gems/scout_apm-2.6.2/lib/scout_apm.rb:201:in `block in <class:Railtie>'
/usr/local/bundle/gems/railties-4.2.11.1/lib/rails/initializable.rb:30:in `instance_exec'
```

Updating to Parser version 2.6.5.0, or downgrading to 2.6.2.x appears to fix this issue.

**Comment by cschneid** [comment]
closing, as this appears to be a parser bug, isolated to only an already outdated version or two. If you run into this, update the Parser gem, and that should fix it. Please let us know if this reappears.

**Comment by dira** [comment]
The context was:

I'm jumping from scout ruby agent 2.4.12 to 2.6.2, which requires `parser` without any version requirements; I'm on  ruby 2.5.3p105, using `rubocop` (0.72.0) in the `development` and `test` environments.
Rubocop requires `parser (>= 2.6)`, and some time ago bundler installed `parser (2.6.3.0)`.

When upgrading the scout agent, the already Gemlock-ed `parser` version satisfied the requirements, was kept. When deploying to the pre-production servers, that had `scout` with `monitor: true`, on `bundle exec rake assets:precompile` I get this error `NameError: uninitialized constant Parser::Base`

Setting `monitor: true` in the development environment, I got the same error.

From my logs, upgrading to `parser 2.6.5` led to the same error.
Regressing to `parser 2.6.2` made it work.
Then I tried again `parser 2.6.5` - and it worked. It also worked on the pre-production servers.

The weird part is that now I cannot make the error happen in the development environment.
After that
